Resources/doc/features/symfony-http-cache.rst

```diff
@@ -65,9 +67,64 @@ dispatcher to directly call the ``HttpCache`` rather than sending an actual
 web request. This is more efficient, and you don't need to configure the server
 IP address.
 
-The :ref:`FOSHttpCache Symfony Proxy Client documentation section <foshttpcache:proxy client symfony httpcache kernel dispatcher>`
-explains how to adjust your bootstrap - you will need to do this in both
-``public/index.php`` and ``bin/console``.
+You need to make your regular kernel aware of the cache kernel::
+
+    // src/AppKernel.php
+
+    namespace App;
+
+    use FOS\HttpCache\SymfonyCache\HttpCacheAware;
+    use FOS\HttpCache\SymfonyCache\HttpCacheProvider;
+    use Symfony\Component\HttpKernel\Kernel;
+
+    class AppKernel extends Kernel implements HttpCacheProvider
+    {
+        use HttpCacheAware;
+        ...
+    }
+
+Then you need to adjust your bootstrap to tell your kernel about the cache.
+Contrary to the regular Symfony documentation, we always create the cache
+kernel so that cache invalidation works in all environments, but only
+actually use the cache kernel in the ``prod`` environment::
+
+    // public/index.php
+
+    use App\AppKernel;
+    use App\CacheKernel;
+
+    // ...
+
+    $kernel = new AppKernel($env, $debug);
+    $cacheKernel = new CacheKernel($kernel);
+    // Tell the kernel about the cache
+    $kernel->setHttpCache($cacheKernel);
+
+    if ('prod' === $env) {
+        $response = $cacheKernel->handle($request);
+    } else {
+        $response = $kernel->handle($request);
+    }
```

Comment:
sorry one question does `terminate` need to be called on $kernel or $cacheKernel or does it make no difference?

Reply:
good point! i took up your suggestion of creating the cache kernel inside the kernel, so much easier this way!